ZenPacks/zenoss/DigMonitor/datasources/DigMonitorDataSource.py

```diff
@@ -42,82 +76,275 @@ class DigMonitorDataSource(ZenPackPersistence, RRDDataSource.RRDDataSource):
+            'device': config.id,
+            'eventClass': ds0.eventClass,
+            'severity': ZenEventClasses.Clear
+        })
 
+        return data
 
-    def getDescription(self):
-        if self.sourcetype == self.DIG_MONITOR:
-            ipAddress = getattr(self, 'ipAddress', '')
-            hostname = getattr(self, 'hostname', '')
-            description = ipAddress + hostname
+    def onError(self, result, config):
+        import pdb; pdb.set_trace()
+        respTime = time.time() - self._startTime
+        DNSNameError = False
+        data = self.new_data()
+        perfData = {}
+        ds0 = config.datasources[0]
+        if hasattr(result.value, 'subFailure'):
+            if isinstance(result.value.subFailure.value, error.DNSNameError):
+                message = ("DNS WARNING - "
+                    "Server: {} - Domain name '{}' "
+                    "not found".format(self.resolver.pickServer()[0],
+                        ds0.params['recordName']))
+                DNSNameError = True
+            elif isinstance(result.value.subFailure.value, defer.TimeoutError):
+                message = ("DNS WARNING - "
+                    "Server: {}, Port: {} - "
+                    "TimeoutError".format(
+                        self.resolver.pickServer()[0],
+                        self.resolver.pickServer()[1]))
+            severity = ds0.severity
         else:
-            description = RRDDataSource.RRDDataSource.getDescription(self)
-        return description
+            message = '{}'.format(result.getErrorMessage())
+            severity = ZenEventClasses.Error
+
+        if not isinstance(result.value, DnsException) and not DNSNameError:
+            respTime = None
+        
+        log.error('{} {}'.format(config.id, message))
+
+        if respTime:
+            perfData['time'] = respTime
+            for dp in ds0.points:
+                if dp.id in perfData:
+                    data['values'][None][dp.id] = perfData[dp.id]
+
+        eventKey = ds0.eventKey or 'DigMonitorWarning'
+        data['events'].append({
+            'eventClassKey': 'DigMonitorWarning',
+            'eventKey': eventKey,
+            'summary': message,
+            'message': message,
+            'device': config.id,
+            'severity': severity,
+            'eventClass': ds0.eventClass
+        })
+
+        return data
+
+    def _getCommonValues(self, response, respTime):
+        s = response.__str__()
+        values = [respTime]
+        for x in ('name', 'ttl', 'class', 'type',):
+            pattern = r'(?<={}=)[^\s]+'.format(x)
+            if x == 'ttl':
+                pattern = r'(?<={}=)[^\s^s]+'.format(x)
+            try:
+                values.append(
+                    re.search(re.compile(pattern), s).group())
+            except AttributeError:
+                values.append(None)
+                
+        return values       
+
+    def messageA(self, response, respTime):
```

Comment:
Probably it is possible to create one function to process all messages, isn't it?

Reply:
no, different records return messages in different formats.

Reply:
Common format is only for CNAME, NS and PTR records.